agent/command/test_selection_get.go

```diff
@@ -92,7 +110,20 @@ func (c *testSelectionGet) Execute(ctx context.Context, comm client.Communicator
 		c.OutputFile = GetWorkingDirectory(conf, c.OutputFile)
 	}
 
-	if !c.isTestSelectionAllowed(conf) {
+	calledAPI := false
+	defer trace.SpanFromContext(ctx).SetAttributes(attribute.Bool(testSelectionCalledAttribute, calledAPI))
```

Comment:
Does this work in staging? I would have thought that this attribute would always resolve to false despite `calledAPI` being set to true below because the arguments are resolved on the line where defer is used, not when the deferred function actually runs. I believe the linter is warning below because the true value is never actually read because the defer has already resolved the value.

Regardless, it's also probably a little to clever to rely on knowing how defer resolves arguments, and Go is a language that prefers things to be explicit. I would recommend setting this explicitly after the value of `calledAPI` is actually known even if it's just to avoid confusion.